### PR TITLE
Streamline namings in EVM-related code

### PR DESCRIPTION
**Pull request by antonyshchenko**
Mostly changing `Blockchain` prefix to `Evm` prefix in types and var names plus some package reorg.

### CODE REVIEW

File: backend/src/main/kotlin/xyz/funkybit/core/blockchain/evm/EvmClient.kt [hunk 1]
```diff
@@ -53,13 +54,33 @@ import xyz.funkybit.core.utils.toHexBytes
 import java.math.BigDecimal
 import java.math.BigInteger
 import kotlin.jvm.optionals.getOrNull
-
+import kotlin.time.Duration
+
+data class EvmClientConfig(
```
Comment (bflood366) [inline backend/src/main/kotlin/xyz/funkybit/core/blockchain/evm/EvmClient.kt:59]: most of this isn't really config associated with the  client but just general EVM configuration for other parts of the system - the only thing in here that the client needs is the URL and maybe the faucet private key but the rest of this is used by other parts of the system.

File: integrationtests/src/test/kotlin/xyz/funkybit/integrationtests/bitcoin/ArchOnboardingTest.kt [hunk 2]
```diff
@@ -3,16 +3,15 @@ package xyz.funkybit.integrationtests.bitcoin
...
 import org.jetbrains.exposed.sql.selectAll
 import org.jetbrains.exposed.sql.transactions.transaction
 import org.junit.jupiter.api.Assertions.assertFalse
 import org.junit.jupiter.api.Assertions.assertTrue
 import org.junit.jupiter.api.Assumptions
 import org.junit.jupiter.api.extension.ExtendWith
-import xyz.funkybit.core.blockchain.ContractType
 import xyz.funkybit.core.blockchain.bitcoin.ArchNetworkClient
 import xyz.funkybit.core.blockchain.bitcoin.BitcoinClient
+import xyz.funkybit.core.blockchain.evm.ContractType
```
Comment (bflood366) [inline integrationtests/src/test/kotlin/xyz/funkybit/integrationtests/bitcoin/ArchOnboardingTest.kt:14]: don't think ContractType is really an evm thing - it is chain agnostic. We have the exchange contract on EVM and non EVM.

**Review by bflood366 — APPROVED** [review]
looks great - nice job

File: backend/src/main/kotlin/xyz/funkybit/core/blockchain/evm/EvmClient.kt (hunk 1 above)
Comment (antonyshchenko) [reply backend/src/main/kotlin/xyz/funkybit/core/blockchain/evm/EvmClient.kt:59]: leaving as is for now, will create a ticket to refactor

File: integrationtests/src/test/kotlin/xyz/funkybit/integrationtests/bitcoin/ArchOnboardingTest.kt (hunk 2 above)
Comment (antonyshchenko) [reply integrationtests/src/test/kotlin/xyz/funkybit/integrationtests/bitcoin/ArchOnboardingTest.kt:14]: good point, moved to `xyz.funkybit.core.blockchain.ContractType `

File: backend/src/main/kotlin/xyz/funkybit/core/blockchain/evm/EvmClient.kt (hunk 1 above)
Comment (antonyshchenko) [reply backend/src/main/kotlin/xyz/funkybit/core/blockchain/evm/EvmClient.kt:59]: Created https://funkybit.atlassian.net/browse/CHAIN-542

**Pull request merged**